### PR TITLE
Admin/security

### PR DESCRIPTION
**Pull request by spyeadon**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -55,6 +55,17 @@
     elsif !BSON::ObjectId.legal?(@token)
       puts 'haha'
       halt(401, "Invalid Token")
+    elsif request.path_info.include? 'admin'
+      puts "Checking admin credentials"
+      session = collection.find( {:_id => BSON::ObjectId(@token) }).first
+      if session.nil?
+        halt(401, "Invalid Token in admin check")
```
Comment (zachgarwood) [inline app.rb]: Lines 60-62 are basically the same as lines 70-73. See if you can find a way to consolidate that functionality and eliminate duplicate code. Check out the Helpers section of the intro docs: http://sinatrarb.com/intro.html.

**Review by zachgarwood — CHANGES_REQUESTED** [review]
See if you can [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) up this code.

**Comment by spyeadon** [comment]
@zachgarwood Made some changes. I did not use the helpers, but I did DRY up the if else logic in the before do so there is no redundancy

**Review by zachgarwood — APPROVED** [review]
Nice as ice!

**Pull request merged**